### ISSUE TITLE
`Error` doesn't work with generic

### ISSUE THREAD

**Issue by KSXGitHub** [JelteF/derive_more#244]
Bug description and steps to reproduce is in this repo: https://github.com/KSXGitHub/derive-more-error-bug-generic

**Comment by JelteF** [comment]
Could you share the actual error? That you get?

**Comment by KSXGitHub** [comment]
```
error: proc-macro derive panicked
 --> src/lib.rs:4:26
  |
4 | #[derive(Debug, Display, Error)]
  |                          ^^^^^
  |
  = help: message: expected `,`

error: could not compile `derive-more-error-bug-generic` due to previous error
```

**Comment by KSXGitHub** [comment]
I also deleted `~/.cargo` and reran `cargo clippy`, same result.

**Comment by ilslv** [comment]
@KSXGitHub try adding `features = ["display", "error"]`

**Comment by KSXGitHub** [comment]
@ilslv Your advice works (with minimal edit to existing code).

Regarding the error message for the old syntax. It is too ambiguous in my opinion: It saids <code>use &#96;"{name}: {source}"&#96; instead</code> but the old syntax already has `"{name}: {source}"`! It took me a while to realize that I need to remove `fmt =`. The error would be better if it tells user what to remove instead of what to include.

**Comment by ilslv** [comment]
@KSXGitHub thanks for your perspective. I've addressed error message in https://github.com/JelteF/derive_more/pull/247

**Comment by tyranron** [comment]
Closing this, as this is already resolved on latest `master` and will be released in 1.0.0.
